sdk/textanalytics/azure-ai-textanalytics/src/main/java/com/azure/ai/textanalytics/AnalyzeTasksAsyncClient.java

```diff
@@ -0,0 +1,390 @@
+        List<ExtractKeyPhrasesResultCollection> keyPhrasesResultCollections = null;
+        if (!CoreUtils.isNullOrEmpty(entityRecognitionTasksItems)) {
+            entitiesResultCollections = entityRecognitionTasksItems.stream()
+                .map(taskItem -> toRecognizeEntitiesResultCollectionResponse(taskItem.getResults()))
+                .collect(Collectors.toList());
+        }
+        if (!CoreUtils.isNullOrEmpty(piiTasksItems)) {
+            piiEntitiesResultCollections = piiTasksItems.stream()
+                .map(taskItem -> toRecognizePiiEntitiesResultCollection(taskItem.getResults()))
+                .collect(Collectors.toList());
+        }
+        if (!CoreUtils.isNullOrEmpty(keyPhraseExtractionTasks)) {
+            keyPhrasesResultCollections = keyPhraseExtractionTasks.stream()
+                .map(taskItem -> toExtractKeyPhrasesResultCollection(taskItem.getResults()))
+                .collect(Collectors.toList());
+        }
+        final AnalyzeTasksResult analyzeTasksResult = new AnalyzeTasksResult(
+            analyzeJobState.getJobId(),
+            analyzeJobState.getCreatedDateTime(),
+            analyzeJobState.getLastUpdateDateTime(),
+            toJobState(analyzeJobState.getStatus()),
+            analyzeJobState.getDisplayName(),
+            analyzeJobState.getExpirationDateTime());
+        AnalyzeTasksResultPropertiesHelper.setErrors(analyzeTasksResult,
+            analyzeJobState.getErrors().stream().map(Utility::toTextAnalyticsError).collect(Collectors.toList()));
+        AnalyzeTasksResultPropertiesHelper.setStatistics(analyzeTasksResult,
+            analyzeJobState.getStatistics() == null ? null : toBatchStatistics(analyzeJobState.getStatistics()));
+        AnalyzeTasksResultPropertiesHelper.setCompleted(analyzeTasksResult, tasksStateTasks.getCompleted());
+        AnalyzeTasksResultPropertiesHelper.setFailed(analyzeTasksResult, tasksStateTasks.getFailed());
+        AnalyzeTasksResultPropertiesHelper.setInProgress(analyzeTasksResult, tasksStateTasks.getInProgress());
+        AnalyzeTasksResultPropertiesHelper.setTotal(analyzeTasksResult, tasksStateTasks.getTotal());
+        AnalyzeTasksResultPropertiesHelper.setEntityRecognitionTasks(analyzeTasksResult, entitiesResultCollections);
+        AnalyzeTasksResultPropertiesHelper.setEntityRecognitionPiiTasks(analyzeTasksResult,
+            piiEntitiesResultCollections);
+        AnalyzeTasksResultPropertiesHelper.setKeyPhraseExtractionTasks(analyzeTasksResult, keyPhrasesResultCollections);
+        return analyzeTasksResult;
+    }
+
+    private Mono<PollResponse<TextAnalyticsOperationResult>> processAnalyzedModelResponse(
+        Response<AnalyzeJobState> analyzeJobStateResponse,
+        PollResponse<TextAnalyticsOperationResult> operationResultPollResponse) {
+
+        LongRunningOperationStatus status;
+        switch (analyzeJobStateResponse.getValue().getStatus()) {
+            case NOT_STARTED:
+            case CANCELLING:
```

Comment:
Just confirming, cancelling == in progress?

Reply:
Cancelling can be consider as in-action operation. There is only in progress that I can use and it works fine.